bin/validate_sample_meta.pl

```diff
@@ -0,0 +1,183 @@
+  $format = 'tsv';
+} else {
+  die("--------\nError: Does not recognise the specified file format, accept values are: 'tsv', 'xls' and 'xlsx'.\n--------\n");
+}
+
+my @in;
+### read in input as array of lines (concatenated columns with tabs)
+if ($format eq 'xlsx'|| $format eq 'xls') {
+  my $book = ReadData ($input);
+  my @rows = Spreadsheet::Read::rows($book->[1]);
+  foreach my $row (@rows) {
+    foreach my $cell (@$row) {
+      $cell = '' if (! defined $cell);
+      # print $cell."\n";
+    }
+    push @in, (join "\t", @$row);
+  }
+} else { # assume it's a tsv
+  open (my $FH, '<', $input) or croak("--------\nError trying to open input file $input: $!\n--------\n");
+  chomp(@in = <$FH>);
+  close $FH;
+}
+
+my ($valid_results, $valid_status) = Sanger::CGP::NgsQc::ValidateSampleMetaAndBamHeader::validate_samples(\@in, $check_all);
+# print "below is the out:\n";
+# foreach my $line (@$valid_results) {print "$line\n";}
+
+### output into the same format again on failed validation, an extra tsv if (input is excel) on a successful validation!
+if ($format eq 'xlsx') {
+  # rename output, if it does not ends with .xlsx
+  if ($output !~ /\.xlsx$/) {
+    $output .= '.xlsx';
+  }
+  my $workbook = Excel::Writer::XLSX->new($output);
+  my $worksheet = $workbook->add_worksheet();
+  my $row = 0;
+  foreach my $line (@$valid_results) {
+    my @array = split("\t", $line);
+    $worksheet->write_row($row, 0, \@array);
+    $row++;
+  }
+  if ($valid_status eq "green") {
+    $output .= ".tsv";
+    warn "\n--------\nSamples validated successfully! Please supply the tsv output file.\ntsv output: $output\n--------\n";
+    open OUT, ">$output" or croak("--------\nCan not open the output: $output to write. $!\n--------\n");
+    foreach my $line (@$valid_results) {print OUT "$line\n";}
+    close OUT;
+  }
+} elsif ($format eq 'xls') {
+  # rename output, if it does not ends with .xls
+  if ($output !~ /\.xls$/) {
+    $output .= '.xls';
+  }
+  my $workbook = Spreadsheet::WriteExcel->new($output);
+  my $worksheet = $workbook->add_worksheet();
+  my $row = 0;
+  foreach my $line (@$valid_results) {
+    my @array = split("\t", $line);
+    $worksheet->write_row($row, 0, \@array);
+    $row++;
+  }
+  if ($valid_status eq "green") {
+    $output .= ".tsv";
+    warn "\n--------\nSamples validated successfully! Please supply the tsv output file.\ntsv output: $output\n--------\n";
+    open OUT, ">$output" or croak("--------\nCan not open the output: $output to write. $!\n--------\n");
+    foreach my $line (@$valid_results) {print OUT "$line\n";}
+    close OUT;
+  }
+} else { # output to tsv format
+  if ($valid_status eq "green") {
+    warn "\n--------\nSamples validated successfully! Please supply the tsv output file.\ntsv output: $output\n--------\n";
+  }
+  open OUT, ">$output" or croak("--------\nCan not open the output: $output to write. $!\n--------\n");
+  foreach my $line (@$valid_results) {print OUT "$line\n";}
```

Comment:
Although `autodie` will check many common errors [it will not test the success of a print statement](http://search.cpan.org/dist/autodie/lib/autodie.pm#print)

```perl
print $OUT $line,"\n" or die $!;
```